cypari2/pari_instance.pyx

```diff
@@ -385,22 +387,30 @@ def prec_words_to_dec(long prec_in_words):
 
 # Callbacks from PARI to print stuff using sys.stdout.write() instead
 # of C library functions like puts().
-cdef PariOUT sage_pariOut
+cdef PariOUT python_pariOut
 
-cdef void sage_putchar(char c):
+cdef void python_putchar(char c):
     cdef char s[2]
     s[0] = c
     s[1] = 0
-    sys.stdout.write(s)
+    try:
+        # avoid string conversion if possible
+        sys.stdout.buffer.write(s)
+    except AttributeError:
+        sys.stdout.write(to_string(s))
```

Comment:
Yes you do. It is perfectly valid to redirect `sys.stdout` to some other streams that might not support the buffer thing (like the doctest module does). In other words the `except` is not only intended for Python2. In particular the `to_string` is mandatory in Python3 (and useless in Python2). Note that `to_string` does not perform any conversion in Python2.